### PR TITLE
Move from DEL to UNLINK

### PR DESCRIPTION
**Pull request by solracsf**
## Summary

TL;DR

The Redis [unlink](https://redis.io/commands/unlink) command is non-blocking and will perform the actual deletion asynchronously.

> UNLINK is a smart command: it calculates the deallocation cost of an object, and if it is very small it will just do what DEL is supposed to do and free the object ASAP. Otherwise the object is sent to the background queue for processing.

http://antirez.com/news/93

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by blizzz — APPROVED** [review]
requires Redis >= 4.0., but that's more than 5 yrs ago. Did not doublecheck with other commands.

**Review by juliusknorr — APPROVED** [review]
Nice, makes sense from reading the docs 👍

**Comment by blizzz** [comment]
failing tests unrelated
